app/src/main/java/de/mathfactory/mooltifill/MooltipassDevice.kt

```diff
@@ -430,16 +432,32 @@ class MooltipassScan {
 
     private fun pairedDevice(context: Context): BluetoothDevice? {
         val bluetoothManager = context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
-        return bluetoothManager.adapter.bondedDevices.lastOrNull(::filter)
+        return bluetoothManager.adapter.bondedDevices
+                .firstOrNull { it -> isConnected(it, context) }
```

Comment:
Example for when this will fail: The user has already bonded and connected a device, then installs the app -> the connection broadcast event will not be seen, because the app hasn't been installed yet, so isConnected will return false for all `bondedDevices`, leading to no connection being made. If there is has not been a device observed yet, I suggest to fallback to the previous method, i.e.: `return bluetoothManager.adapter.bondedDevices.lastOrNull(::filter)`

Reply:
In this case if we don't have any last connected device yet on device, we can just check following conditions

- if device is bonded and device has same mac address, in this case we don't check last connected state right?
-

Reply:
"[...] the device has the same mac address, [...]" - "same" compared to what? In the case we don't have any last connected device address stored, we don't have anything to compare to, do we?

Reply:
Yes, I have already noticed it and covered it. You will get in the next commit.